File: ai/mcp/python/clients/MathJsMath.py
from typing import Optional, Any, List, Union

from ..McpClient import McpClient

# MathJs Math Expression Evaluator.
class MathJsMath(McpClient):
    """
    MathJs Math Expression Evaluator.
    """
    def __init__(self):
        super().__init__()

    async def callMathExpressionEvaluatorTool(self, expression: str) -> Union[Any, None]:
        """
        call the math expression evaluator tool.

        Args:
            expression: the math expression.

        Return:
            the evaluated expression.
        """
        res: Any = await self.callTool("MathExpressionEvaluator", args={"expression": expression})

        # return the result.
        return res;

    async def callMathExpressionEvaluatorPrompt(self, expression: str) -> Union[Any, None]:
        """
        call the math expression evaluator prompt.

        Args:
            expression: the math expression.

        Return:
            the evaluated expression.
        """
        res: Any = await self.callPrompt("MathExpressionEvaluator", args={"expression": expression})

        # return the result.
        return res;

    async def callMathExpressionResultPrompt(self, result: str) -> Union[Any, None]:
        """
        call the math expression result prompt.

        Args:
            result: the math result.

        Return:
            the evaluated result.
        """
        res: Any = await self.callPrompt("MathExpressionResult", args={"result": result})

        # return the result.
        return res;

    async def callMathJsDocsUrlResource(self) -> Union[Any, None]:
        """
        call the MathJs documentation URL resource.

        Return:
            the resource result.
        """
        res: Any = await self.callResource("mathjs://{version}")

        # return the result.
        return res;

    async def callMathJsDocsUrlVsersionResource(self, version: str) -> Union[Any, None]:
        """
        call the MathJs documentation URL resource.

        Args:
            version: the version.

        Return:
            the resource result.
        """
        res: Any = await self.callResource(f"mathjs://doc/{version}/num")

        # return the result.
        return res;

File: ai/mcp/typescript/clients/MathJsMath.ts
import { McpClient } from '../McpClient.js';

/**
 * MathJs Math Expression Evaluator.
 */
export class MathJsMathClient extends McpClient {

    /**
     * MathJs Math Expression Evaluator.
     */
    constructor() {
        super("MathJsMathExpressionEvaluator", "1.0.1");
    }

    /**
     * call the math expression evaluator tool.
     * @param {string} expression     the math expression.
     * @returns {Promise<any>} the evaluated expression.
     */
    async callMathExpressionEvaluatorTool(expression: string): Promise<any> {
        let res: any = await this.callTool("MathExpressionEvaluator", {
            expression: expression
        });

        // return the result.
        return res;
    }

    /**
     * call the math expression evaluator prompt.
     * @param {string} expression     the math expression.
     * @returns {Promise<any>} the evaluated expression.
     */
    async callMathExpressionEvaluatorPrompt(expression: string): Promise<any> {
        let res: any = await this.callPrompt("MathExpressionEvaluator", {
            expression: expression
        });

        // return the result.
        return res;
    }

    /**
     * call the math expression result prompt.
     * @param {string} result     the math result.
     * @returns {Promise<any>} the evaluated result.
     */
    async callMathExpressionResultPrompt(result: string): Promise<any> {
        let res: any = await this.callPrompt("MathExpressionResult", {
            result: result
        });

        // return the result.
        return res;
    }

    /**
     * call the MathJs documentation URL resource.
     * @returns {Promise<any>} the evaluated result.
     */
    async callMathJsDocsUrlResource(): Promise<any> {
        let res: any = await this.callResource("mathjs://{version}");

        // return the result.
        return res;
    }

    /**
     * call the MathJs documentation URL resource.
     * @param {string} version     the version.
     * @returns {Promise<any>} the evaluated result.
     */
    async callMathJsDocsUrlVersionResource(version: string): Promise<any> {
        let res: any = await this.callResource(`mathjs://doc/${version}/num`);

        // return the result.
        return res;
    }
}